### DISCUSSION TITLE
Enabling Plasma/THC for testing (newby alert)

### DISCUSSION THREAD

**Discussion by sntlewis** [grblHAL/iMXRT1062#65] [Q&A]
Cross posted to "core" discussion as well...

Ok a few disclaimers...

rank newby to GRBL/GRBLHal world..
very familiar with Marlin/MACH3 (built several 3d printers and cnc routers)
very early in my next "build." This will be a CNC Plasma table. I already have the design/build underway albeit slow due to $$$.
understand that GRBLHal Plasma is young and under development.
Now for the question of the day as I dig into/learn GRBLHal. Other than modifying my.machine.h to enable "plasma" what else is needed to be done before compile/flash. My MCU is the Teensy 4.1 with the T41U5XBB BOB.

Terje noted a "daughter board" from Phil is needed.  Already reached out to him via Tindie.

**Comment by sntlewis** [comment]
I would have thought that if I set PLASMA_ENABLE in my_machine.h and then compile/build.  I should see a panel for Plasma/THC on the IOSender when connected to the T41U5XXBB...

Any guidance would be appreciated.

**Reply by terjeio** [reply]
> I should see a panel for Plasma/THC on the IOSender when connected to the T41U5XXBB...

The plugin does not initialize itself unless an analog input (and tree digital inputs) is available. I need to remove the analog input dependency since it can be configured to work with up/down signals instead. I'll do that soon.

**Reply by sntlewis** [reply]
doesn't the analog signal provide the plasma voltage (suitably divided to prevent blowing out the opto, likely 50:1) so that the THC control can adjust height?

**Reply by terjeio** [reply]
> doesn't the analog signal provide the plasma voltage (suitably divided to prevent blowing out the opto, likely 50:1) so that the THC control can adjust height?

Correct, but the control part is work in progress and may not work. I have some new ideas for the feedback handling (injecting steps for the Z-axis) - but not enough time for everything...

I have now committed an update that should allow the plugin to initialize with only 3 digital inputs or 1 digital and 1 analog.

**Reply by sntlewis** [reply]
1 digital input and 1 analog input would give the voltage out from the plasma (divided down to 5v), input to the teensy/bob as well as the arc ok? output from the plasma.  I could then use spindle on/off to trigger the arc start/stop?  Still looking into the mcp3221 i2c card that @Phil-Barrett worked on as well as it would add another layer of safety/isolation to the system.
